feat(maritalk): update models and enhance component

viniciossilva3 opened the pull request:
- Update model options to sabiazinho-3 and sabia-3
- Replace FloatInput with SliderInput for temperature control
- Add informative descriptions for model selection
- Add system_message parameter to ChatMaritalk
- Make api_key field required

Modified files:
- [src/backend/base/langflow/components/models/maritalk.py](url)

viniciossilva3 committed:
feat(maritalk): update models and enhance component

- Update model options to sabiazinho-3 and sabia-3
- Replace FloatInput with SliderInput for temperature control
- Add informative descriptions for model selection
- Add system_message parameter to ChatMaritalk
- Make api_key field required

Modified files:
- src/backend/base/langflow/components/models/maritalk.py

Co-authored-by: Vinícios Batista da Silva <vinicios.batsi@gmail.com>

viniciossilva3 committed:
feat(maritalk): update models and enhance component

- Update model options to sabiazinho-3 and sabia-3
- Replace FloatInput with SliderInput for temperature control
- Add informative descriptions for model selection
- Add system_message parameter to ChatMaritalk
- Make api_key field required

Modified files:
- src/backend/base/langflow/components/models/maritalk.py

Co-authored-by: Vinícios Batista da Silva <vinicios.batsi@gmail.com>

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

ogabrielluiz commented on src/backend/base/langflow/components/models/maritalk.py:
```diff
@@ -24,29 +24,37 @@ class MaritalkModelComponent(LCModelComponent):
             advanced=False,
-            options=["sabia-2-small", "sabia-2-medium"],
-            value=["sabia-2-small"],
+            info="Choose between Sabiazinho-3 (smaller and faster) or Sabia-3 (larger and more capable) models.",
+            options=["sabiazinho-3", "sabia-3"],
+            value="sabiazinho-3",
```
What do you think of allowing users to customize the model?
```suggestion
            info="Choose between Sabiazinho-3 (smaller and faster) or Sabia-3 (larger and more capable) models.",
            options=["sabiazinho-3", "sabia-3"],
            value="sabiazinho-3",
            combobox=True
```

viniciossilva3 replied:
Hi @ogabrielluiz , I can do it like Hugging Face too, some models available and a custom field to add as the user wants. Could it be like this? This way it maintains good usability without making it difficult to add other models.

viniciossilva3 replied:
https://github.com/user-attachments/assets/396f284d-3d14-4ac5-b8bb-48e9f45c08df


I modified the component, now it has the following functionalities:

- 2 models by default
- If there is an API key, it searches for models through the API and adds them to the dropdown
- It has a custom field that allows the user to add models as desired.

I will do commit the changes.

viniciossilva3 replied:
https://github.com/user-attachments/assets/65823cb3-ee99-4d1e-9f7f-7384620cdadf

Thank you, I had never used the combobox, I liked the functionality!

Should I leave it with just the API?
and without models by default? And if you want, you can use additional ones manually.

Like in the attached video?

ogabrielluiz replied:
No, we should avoid relying on external systems. Always have a fallback.

viniciossilva3 replied:
Thanks, I will modify it!

viniciossilva3 committed:
feat(maritalk): enhance model fetching and component configuration

- Add dynamic model fetching from Maritaca API with fallback to default models
- Implement real-time model list updates based on API key availability
- Add custom model support with conditional field visibility
- Improve error handling and HTTP status checks
- Add logging for model building operations
- Optimize code structure following linting guidelines

Modified files:
- src/backend/base/langflow/components/models/maritalk.py

Co-authored-by: Vinícios Batista da Silva <vinicios.batsi@gmail.com>

viniciossilva3 committed:
refactor(maritalk): add combobox tag to model dropdown

- Add combobox=True to model dropdown for free text input
- Remove default models in favor of API-provided options
- Remove separate custom model input field
- Update model fetching to return empty list when no API key
- Simplify build configuration update logic
- Add validation for empty model name

Modified files:
- src/backend/base/langflow/components/models/maritalk.py

Co-authored-by: Vinícios Batista da Silva <vinicios.batsi@gmail.com>

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
fix(maritalk): add default model fallback when API is unavailable

- Add DEFAULT_MODELS constant with sabiazinho-3 and sabia-3 as fallback options
- Set default models as initial options for model_name dropdown
- Enhance fetch_models() to return default models when API key is missing or request fails
- Improve update_build_config() to always include default models in options list
- Add detailed error logging for API failures and configuration issues
- Ensure component always has valid model options regardless of API availability

Modified files:
- src/backend/base/langflow/components/models/maritalk.py

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

viniciossilva3 committed:
style(maritalk): fix linting issues and resolve merge conflicts

viniciossilva3 committed:
chore: resolve conflict with Search agent.json by using main version

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

edwinjosechittilappilly committed:
Merge branch 'main' into feat/maritaca-model-update

edwinjosechittilappilly commented:
Testing!

edwinjosechittilappilly committed:
Merge branch 'main' into feat/maritaca-model-update

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

edwinjosechittilappilly committed:
Merge branch 'main' into feat/maritaca-model-update

edwinjosechittilappilly reviewed (changes requested):
@viniciossilva3  Please add test also. Following a standard pattern. refer:
https://github.com/langflow-ai/langflow/blob/main/src/backend/tests/unit/components/models/test_language_model_component.py

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
test: add tests for MaritalkModelComponent

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

viniciossilva3 committed:
Fix: Resolve ARG002 Ruff error in test_maritalk_component.py

viniciossilva3 committed:
Merge branch 'main' into feat/maritaca-model-update

edwinjosechittilappilly committed:
Merge branch 'main' into feat/maritaca-model-update
